core/templates/dev/head/galleries/Gallery.js

```diff
@@ -95,16 +95,32 @@ oppia.controller('Gallery', [
     });
 
     $scope.allExplorationsInOrder = [];
+    $scope.allCollectionsInOrder = [];
```

Comment:
Explorations and collections should be listed together, I think. It's not yet clear whether they should have the same search index, so the merging should probably be done here. I recommend storing them in one list and specifying an ``activityType`` key to distinguish between an exploration and collection. Figuring out the order is also a bit hard; I think for now just appending the collections to the activity list and pushing the explorations to be listed first should be fine, unless you have other thoughts on that.